✨ #175 Ability to disable Haptic Feedback

cyborg-joshi opened the pull request:
The changes made in commit "Ability to disable Trackpad Haptic Feedback #175" allow users to control whether Trackpad Haptic Feedback is enabled or disabled through a new setting.

cyborg-joshi committed:
Implemented the ability to disable Trackpad Haptic Feedback in response to issue

MrKai77 commented:
Looks good so far! Would you also be able to add a toggle button in the "More" tab of the settings window to enable/disable this, under a new section called "Accessibility"?

Also, I think renaming `enableTrackpadHapticFeedback` to `enableHapticFeedback` would be better too :)

cyborg-joshi committed:
#176 Refactored SwiftUI code in MoreSettingsView

cyborg-joshi commented:
Thank you for your feedback, @MrKai77 ! I've implemented the requested changes:
- Added a toggle button in the "More" tab under a new section called "Accessibility" to enable/disable the feature.
- Renamed `enableTrackpadHapticFeedback` to `enableHapticFeedback` for clarity.

<img width="759" alt="image" src="https://github.com/MrKai77/Loop/assets/91533278/08fe23a1-b3db-409d-a8fb-6ede9a79d27c">

mrkai77 commented on Loop/Settings/MoreSettingsView.swift:
```diff
@@ -37,8 +37,7 @@ struct MoreSettingsView: View {
                         }, label: {
-                            // swiftlint:disable:next line_length
-                            Text("Current version: \(Bundle.main.appVersion) (\(Bundle.main.appBuild)) \(Image(systemName: "doc.on.clipboard"))")
+                            Text("Current version: \(Bundle.main.appVersion) (\(Bundle.main.appBuild))")
```
Please revert this change; not related to this PR

mrkai77 replied:
Still to be fixed :)

mrkai77 commented on Loop/Settings/MoreSettingsView.swift:
```diff
@@ -139,8 +138,17 @@ struct MoreSettingsView: View {
+
+
+
+
```
Remove blank newlines at end of file

mrkai77 commented on Loop/Settings/MoreSettingsView.swift:
```diff
@@ -139,8 +138,17 @@ struct MoreSettingsView: View {
+
+            Section(header: Text("Accessibility")) {
+                Toggle("Enable Haptic Feedback", isOn: $enableHapticFeedback)
+            }
```
Looks good!
I'd recommend using `Section("Accessibility")` :)
Also, now that I'm looking at this, I think it would be better if this section was in between the Stage Manager section and the Permissions section.

cyborg-joshi replied:
<img width="616" alt="image" src="https://github.com/MrKai77/Loop/assets/91533278/6cc969d1-2f14-40a0-a69e-c683818bda9f">
@MrKai77 
I've made the recommended changes, including using Section("Accessibility") and reordering the sections to place "Accessibility" between "Stage Manager" and "Permissions" for better organization. Please take a look and let me know if there's anything else you'd like to adjust or if you have any additional feedback.

MrKai77 commented:
@cyborg-joshi any updates on this? I'm planning to make a release soon, and I'd love to have this in it :)

cyborg-joshi commented:
@MrKai77 > * Added a toggle button in the "More" tab under a new section called "Accessibility" to enable/disable the feature

and made the commit too

MrKai77 commented:
Oh @cyborg-joshi I think you may have forgotten to read the code review? I added some feedback :)

cyborg-joshi committed:
Merge branch 'develop' into cyborg_joshi-#175

MrKai77 commented:
I'll convert this to a draft for now as it is not ready for merge.

MrKai77 commented:
@cyborg-joshi any updates? Let me know if you need help getting to the code review :)

cyborg-joshi commented:
@MrKai77   thank you for the prompt. I'd appreciate a bit more clarity on the specific changes or updates you're looking for in the code review. Could you please provide a brief summary or point out the areas that need attention? This will help me address your feedback more effectively. Thanks!

MrKai77 commented:
@cyborg-joshi I'm so sorry! It seems as if I forgot to submit the code review 😅
Now that I have submitted it, you should be able to view it above, and make the needed changes :)

cyborg-joshi committed:
Merge branch 'develop' into cyborg_joshi-#175

cyborg-joshi commented:
@MrKai77  made the necessary changes

MrKai77 commented:
@cyborg-joshi  this looks good! But it seems like you committed the changes to your `develop` branch rather than `cyborg_joshi-#175`, so those changes aren't currently visible in this PR. Would you be able to recommit that in the correct branch? Thanks!

cyborg-joshi committed:
#176 Added an "Accessibility" section with a toggle for "Enable Haptic Feedback"

cyborg-joshi commented:
@MrKai77  Sure, I'll recommit the changes to the correct branch (cyborg_joshi-#175). Thank you for pointing that out!

mrkai77 commented on Loop/Managers/LoopManager.swift:
```diff
@@ -129,12 +129,14 @@ class LoopManager: ObservableObject {
+                    performanceTime: NSHapticFeedbackManager.PerformanceTime.now
+                   )
+               }
+        
```
Fix indentation

cyborg-joshi committed:
#176

cyborg-joshi commented:
@MrKai77 I've made the changes to the MoreSettingsView.swift file. Are these changes okay, or did I misunderstand the modifications? Please let me know if any further adjustments are needed.

mrkai77 commented on Loop/Settings/MoreSettingsView.swift:
```diff
@@ -37,9 +37,12 @@ struct MoreSettingsView: View {
+                                    .font(.caption)
+                                    .foregroundColor(.secondary)
+                                Image(systemName: "doc.on.clipboard")
+                            }
```
Use 
```swift
// swiftlint:disable:next line_length
Text("Current version: \(Bundle.main.appVersion) (\(Bundle.main.appBuild)) \(Image(systemName: "doc.on.clipboard"))")
```

MrKai77 commented:
Yup, the changes you made to `MoreSettingsView.swift` are perfect, except for the part where you changed the clipboard icon in the text that shows the version number. It should be this: 
```swift
Text("Current version: \(Bundle.main.appVersion) (\(Bundle.main.appBuild)) \(Image(systemName: "doc.on.clipboard"))")
```
This way, all the modifiers that are applied to the text are also applied to the icon.

As for the "Fix indentation" comment, you can clearly see that your `if` statement is off by one indent, as well as that there are trailing spaces on the lines, as shown below:
![Screenshot 2024-01-30 at 1 24 46 AM](https://github.com/MrKai77/Loop/assets/68963405/b143af03-2cae-4282-9c31-1438cb1161b7)

EDIT: Actually, since these are very minor changes, if you'd like me to do them for you, let me know!

cyborg-joshi committed:
#176  Fixed identation and trailing errors

cyborg-joshi commented:
@MrKai77 Thank you for providing the details. I've made the adjustments to the 'MoreSettingsView.swift' file according to your specifications. 
I also reviewed the 'Fix indentation' comment and addressed the issue by adjusting the indentation and removing trailing spaces. Please have a look and let me know if everything meets your expectations.
if still any minor changes
please do it for me

MrKai77 committed:
Merge branch 'develop' into cyborg_joshi-#175

MrKai77 commented:
Looks good!
